util/types/json/json.go

```diff
@@ -15,26 +15,130 @@ package json
 
 import (
 	"bytes"
+	"encoding/json"
+	"strings"
+	"unsafe"
 
 	"github.com/pingcap/tidb/mysql"
 	"github.com/pingcap/tidb/terror"
+	"github.com/pingcap/tidb/util/hack"
 )
 
-// JSON is for MySQL JSON type.
-type JSON interface {
-	getTypeCode() byte
-	encode(buffer *bytes.Buffer)
-	// String returns JSON text.
-	String() string
-	// Type returns type string.
+const (
+	typeCodeObject  byte = 0x01
+	typeCodeArray   byte = 0x03
+	typeCodeLiteral byte = 0x04
+	typeCodeInt64   byte = 0x09
+	typeCodeFloat64 byte = 0x0b
+	typeCodeString  byte = 0x0c
+)
+
+const (
+	jsonLiteralNil   byte = 0x00
+	jsonLiteralTrue  byte = 0x01
+	jsonLiteralFalse byte = 0x02
+)
+
+const internalErrorUnknownTypeCode = "unknown type code"
+
+var _ AccessMethod = JSON{}
+
+// AccessMethod is for MyQL JSON functions.
```

Comment:
it's not required. I use it just as a method list, so we can know which access methods JSON has more easily.  Is that a bad style?